### PR TITLE
feat(issue-platform): adds a partition key to issue platform producer

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a setting called `issue_platform.use_kafka_partition_key` that when enabled will set a partition key when we produce to the issue platform consumer. We partition by the first fingerprint that we find. Note that technically there could be multiple fingerprints but I believe we can just use the first fingerprint.

### CODE REVIEW

File: src/sentry/issues/producer.py [hunk 1]
```diff
@@ -64,7 +65,14 @@ def produce_occurrence_to_kafka(
...
 
-    payload = KafkaPayload(None, json.dumps(payload_data).encode("utf-8"), [])
+    partition_key = None
+    if (
+        options.get("issue_platform.use_kafka_partition_key")
+        and occurrence
+        and occurrence.fingerprint
+    ):
+        partition_key = bytes(occurrence.fingerprint[0], "utf-8")
+    payload = KafkaPayload(partition_key, json.dumps(payload_data).encode("utf-8"), [])
```
Comment (JoshFerge) [inline src/sentry/issues/producer.py:75]: are we worried about a single fingerprint overwhelming one of the consumers? could that be possible?

**Review by JoshFerge — APPROVED** [review]
makes sense to me!

File: src/sentry/issues/producer.py (hunk 1 above)
Comment (scefali) [reply src/sentry/issues/producer.py:75]: @JoshFerge I don't know, let me check with the experts

Comment (wedamija) [reply src/sentry/issues/producer.py:75]: It's a possibility but I don't think it will be a problem in practice

**Pull request merged**